src/lib/cj/client.rs

```diff
@@ -11,31 +11,43 @@ pub struct CJS2SClient {
     cj_cid: String,
     cj_type: String,
     cj_signature: String,
+    random_minutes: Duration,
+}
+
+fn get_random_minutes() -> Duration {
+    let mut rng = thread_rng();
+    let minutes = rng.gen_range(15..=60);
+    Duration::minutes(minutes)
 }
 
 impl CJS2SClient {
-    pub fn new(settings: &Settings, cj_endpoint: Option<&str>) -> CJS2SClient {
+    pub fn new(
+        settings: &Settings,
+        cj_endpoint: Option<&str>,
+        random_minutes: Option<Duration>,
+    ) -> CJS2SClient {
         let cj_endpoint = cj_endpoint.unwrap_or("https://www.emjcd.com/u");
         CJS2SClient {
             url: Url::parse(cj_endpoint).expect("Could not parse cj_endpoint"),
             client: Client::new(),
             cj_cid: settings.cj_cid.clone(),
             cj_type: settings.cj_type.clone(),
             cj_signature: settings.cj_signature.clone(),
+            random_minutes: random_minutes.unwrap_or_else(get_random_minutes),
```

Comment:
What is the purpose of passing this in here if we are just going to fall back to a random function?